src/btreemap.rs

```diff
@@ -3071,4 +3120,137 @@ mod test {
 
         assert_eq!(btree.allocator.num_allocated_chunks(), 0);
     }
+
+    #[test]
+    fn first_key_value_correctness() {
+        btree_test(|mut btree| {
+            // Initially, the tree is empty.
+            assert_eq!(btree.first_key_value(), None);
+
+            // Insert some elements.
+            assert_eq!(btree.insert(b(&[3]), b(&[30])), None);
```

Comment:
It'd be better to write a few loops to check the invariants of each case (insert + read, remove + read, pop + read), eg

```rs
let n = 100;
let key = |i| b(&[i]);
let value = |i| b(&[i * 10]);

// insert + read
for i in 0..n {
  assert_eq!(btree.insert(key(i), value(i)), None);
  assert_eq!(btree.first_key_value(), Some(key(0)));
  assert_eq!(btree.last_key_value(), Some(key(i)));
}

// remove + read
for i in 0..n {
  assert_eq!(btree.remove(key(i), value(i)), Some(value(i)));
  assert_eq!(btree.first_key_value(), Some(key(i+1)));
}

...
```